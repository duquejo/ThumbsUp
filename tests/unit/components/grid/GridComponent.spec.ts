import { beforeEach, describe, expect, it, vi } from 'vitest';
import { VueWrapper, shallowMount } from '@vue/test-utils';
import { createTestingPinia } from '@pinia/testing';
import GridComponentVue from '@/components/grid/GridComponent.vue';
import CardComponentVue from '@/components/card/CardComponent.vue';
import SelectComponentVue from '@/components/select/SelectComponent.vue';

describe('+ HeroComponent unit tests', () => {

  let wrapper: VueWrapper;

  beforeEach(() => {
    vi.clearAllMocks();
    wrapper = shallowMount(GridComponentVue, {
      global: {
        plugins: [createTestingPinia({
          createSpy: vi.fn(),
        })],
      },
    });
  });

  it('should render as expected', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render expected child components', () => {
    expect(wrapper.findComponent(SelectComponentVue).exists()).toBeTruthy();
    expect(wrapper.findAllComponents(CardComponentVue).length).toBeGreaterThan(0);
  });
});
