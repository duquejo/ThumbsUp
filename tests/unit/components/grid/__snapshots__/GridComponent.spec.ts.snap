// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`+ HeroComponent unit tests > should render a default text of there are no celebrities loaded 1`] = `
"<main data-v-0dada1a8="" role="main">
  <div data-v-0dada1a8="" class="main__header"><span data-v-0dada1a8="" class="main__header-title">Previous Rulings</span>
    <div data-v-0dada1a8="" class="main__header-select">
      <select-component-stub data-v-0dada1a8="" options="list,grid" default="grid" tabindex="0"></select-component-stub>
    </div>
  </div>
  <p data-v-0dada1a8="">There aren't enough celebrities... soon.</p>
</main>"
`;

exports[`+ HeroComponent unit tests > should render as expected 1`] = `
"<main data-v-0dada1a8="" role="main">
  <div data-v-0dada1a8="" class="main__header"><span data-v-0dada1a8="" class="main__header-title">Previous Rulings</span>
    <div data-v-0dada1a8="" class="main__header-select">
      <select-component-stub data-v-0dada1a8="" options="list,grid" default="grid" tabindex="0"></select-component-stub>
    </div>
  </div>
  <ol data-v-0dada1a8="" class="main__container">
    <card-component-stub data-v-0dada1a8="" celebrity="[object Object]" islayered="false"></card-component-stub>
    <card-component-stub data-v-0dada1a8="" celebrity="[object Object]" islayered="false"></card-component-stub>
    <card-component-stub data-v-0dada1a8="" celebrity="[object Object]" islayered="false"></card-component-stub>
    <card-component-stub data-v-0dada1a8="" celebrity="[object Object]" islayered="false"></card-component-stub>
    <card-component-stub data-v-0dada1a8="" celebrity="[object Object]" islayered="false"></card-component-stub>
    <card-component-stub data-v-0dada1a8="" celebrity="[object Object]" islayered="false"></card-component-stub>
    <card-component-stub data-v-0dada1a8="" celebrity="[object Object]" islayered="false"></card-component-stub>
  </ol>
</main>"
`;
